Env instructions

samgdotson opened the pull request:
This PR 
* closes #58 by adding an `environment.yml` file and instructions to install it the environment in the `README` and
* closes #57 by adding a Binder badge that links to the example notebooks. There are ways to have the examples in the top-level directory ([example](https://github.com/PyPSA/PyPSA)), such as with the `nbsphinx-link` package, but I could not get it to work properly. 

Lastly, this PR addresses the request from @fredshone to have a notebook that emphasizes the multi-objective aspects of `osier` by creating a new notebook called `getting-started` which serves as a "minimum working example."

samgdotson committed:
adds environment file

samgdotson committed:
removes maximum python version, reruns example

samgdotson committed:
updates gitignore

samgdotson committed:
adds nbsphinx-link

samgdotson committed:
Merge branch 'main' of github.com:arfc/osier into env-instructions

samgdotson committed:
fixes unit bug in land_use_intensity

samgdotson committed:
adds new example notebook

samgdotson committed:
adds new notebook and osier to environment file

samgdotson committed:
updates env file and new notebook

samgdotson committed:
updates environment.yml file

samgdotson committed:
adds conda env instructions and binder badge

samgdotson committed:
adds examples instruction to the readme

nsryan2 commented on docs/source/examples/capacity_expansion_tutorial.ipynb:
Everything makes sense and the variables are good, I think you have a good amount of explanation and useful in-line comments. 

1. I'm noticing is that you aren't adding spaces between comma separated elements or in-line operations, might be good for readability.

E.g., in the Creating the demand profile section you have some definitions that might seem a little dense
`demand = (np.sin((hours*np.pi/n_hours*2+phase_shift))*-1+np.ones(N)*(base_shift+1))`,
and a much less consequential example
`hours = np.linspace(0,N,N)`.

2. In the `Setting up a Pymoo algorithm` section, you mix endings for your list and are inconsistent throughout the notebook between lists.

3. Also in the `Setting up a Pymoo algorithm` section, I am confused by what`The rest is extra.` means at the start.

4. Your box plot at the very end under the `Visualizing Multi-objective Results` section has an error saying the method you used to name the boxes will be depreciated soon, have you tried using `tick_labels` instead of `labels` as they suggest?

nsryan2 commented on docs/source/examples/getting_started.ipynb:
Again, no comments on the code itself. The style is good, very bare-bones (which it seems was the goal). A couple of comments:
1. You make use of this tag `%%time` in this notebook but no others, is this something you want to include in `capacity_expansion_tutorial.ipynb` as well as you incorporate some notion of the wall time of runs in that notebook?
2. Similar comment about the box plot `labels` vs `tick_labels` and avoiding the error where possible.

nsryan2 reviewed (approved):
LGTM! Some minor comments about consistency in the notebooks, but the bulk of these files are simple changes that look good.

samgdotson committed:
Merge branch 'main' of github.com:arfc/osier into env-instructions

samgdotson committed:
adds binder badge to all notebooks

samgdotson commented:
@nsryan2 thanks for the review. I updated the `tick_labels` in the box plots. I also added a Binder badge to every notebook.

If you're happy, please merge!
